project/data_analysis/python/split_nulls/get_split_covariance_blocks.py

```diff
@@ -0,0 +1,204 @@
+"""
+This script compute the analytical covariance matrix elements.
+"""
+import sys
+import numpy as np, pylab as plt
+from pspipe_utils import pspipe_list, best_fits
+
+from pspy import pspy_utils, so_cov, so_dict, so_map, so_mcm, so_mpi, so_spectra
+from itertools import combinations_with_replacement as cwr
+
+d = so_dict.so_dict()
+d.read_from_file(sys.argv[1])
+
+windows_dir = "windows"
+mcms_dir = "mcms"
+spectra_dir = "spectra"
+noise_dir = "split_noise"
+cov_dir = "split_covariances"
+bestfit_dir = "best_fits"
+sq_win_alms_dir = "sq_win_alms"
+
+pspy_utils.create_directory(cov_dir)
+surveys = d["surveys"]
+binning_file = d["binning_file"]
+lmax = d["lmax"]
+niter = d["niter"]
+binned_mcm = d["binned_mcm"]
+apply_kspace_filter = d["apply_kspace_filter"]
+cov_T_E_only = d["cov_T_E_only"]
+
+spectra = ["TT", "TE", "TB", "ET", "BT", "EE", "EB", "BE", "BB"]
+spin_pairs = ["spin0xspin0", "spin0xspin2", "spin2xspin0", "spin2xspin2"]
+
+# fast_coupling is designed to be fast but is not for general usage
+# In particular we assume that the same window function is used in T and Pol
+fast_coupling = True
+
+arrays, n_splits, bl_dict = {}, {}, {}
+ps_all, nl_all = {}, {}
+spec_name = []
+
+l_cmb, cmb_dict = best_fits.cmb_dict_from_file(bestfit_dir + "/cmb.dat", lmax, spectra)
+
+arrays = {sv: d[f"arrays_{sv}"] for sv in surveys}
+array_list = [f"{sv}_{ar}" for sv in surveys for ar in arrays[sv]]
+
+l_fg, fg_dict = best_fits.fg_dict_from_files(bestfit_dir + "/fg_{}x{}.dat", array_list, lmax, spectra)
+
+for sv in surveys:
+    arrays[sv] = d[f"arrays_{sv}"]
+    for ar in arrays[sv]:
+        l_beam, bl_dict[sv, ar] = pspy_utils.read_beam_file(d[f"beam_{sv}_{ar}"])
+        id_beam = np.where((l_beam >= 2) & (l_beam < lmax))
```

Comment:
I'm not sure the `where` is needed here. I think only a slice selection is needed just like in the other files